lib/models/organisation.js

```diff
@@ -210,108 +258,103 @@ schema.statics.limitIdsByOrg = function limitIdsByOrg(
+      organisation: orgModel._id,
+      roles: [adminRole._id],
+    }]
+  );
 
-    const owner = await User.findById(objectId(organisation.owner));
-    if (owner) {
-      // add this org to the creator's allowed orgs
-      const newRoles = [createAdminRole._id];
+  await owner.save();
+};
 
-      const existingOrganisationSettings = owner.organisationSettings || [];
-      let updatedSettings = existingOrganisationSettings;
+const createDependencies = async (orgModel, User) => {
+  const adminRole = await createRoles(orgModel);
+  await updateOwner(orgModel, User, adminRole);
+};
 
-      // check if the user has settings for this org
-      const existingOS = find(existingOrganisationSettings, setting =>
-         setting.organisation.toString() === organisation._id.toString()
-      );
+const removeDependencies = async (orgModel) => {
+  const Role = getConnection().model('Role');
+  const User = getConnection().model('User');
 
-      if (existingOS) {
-        const updatedOS = existingOS;
-        // if the setting already exists, update it
-        if (updatedOS.roles) {
-          // union the roles if exist
-          updatedOS.roles = union(updatedOS.roles, newRoles);
-        } else {
-          // set the new roles if no previous roles exist
-          updatedOS.roles = newRoles;
-        }
-
-        // loop through to create the updated settings array
-        updatedSettings = map(updatedSettings, (os) => {
-          if (os.organisation.toString() === organisation._id.toString()) {
-            // return the updated settings into the array
-            return updatedOS;
-          }
-        });
-      } else {
-        // insert a new item
-        updatedSettings.push({
-          organisation: organisation._id,
-          roles: newRoles
-        });
+  await Role.deleteMany({ organisation: [orgModel._id] });
+  await User.updateMany(
+    {},
+    {
+      $pull: {
+        organisations: orgModel._id,
+        organisationSettings: { organisation: orgModel._id },
       }
+    }
+  );
+};
 
-      // update the settings on the owner
-      owner.organisations = union(owner.organisations, [organisation._id]);
-      owner.organisationSettings = updatedSettings;
 
-      await owner.save();
-    }
+schema.post('init', function handleInit() {
+  this._original = this.toObject();
+});
+
+schema.pre('save', async function preSave(next) {
+  const User = getConnection().model('User');
+
+  if (!isUndefined(this._oldExpiration) && this.expiration !== this._oldExpiration) {
+    this.expirationNotifications.expirationNotificationSent = EMAIL_NOOP;
+    this.expirationNotifications.weekBeforeNotificationSent = EMAIL_NOOP;
+  }
 
-    await Promise.all([createAdminRole, createObserverRole]);
+  if (!isUndefined(this._oldTimezone) && this.timezone !== this._oldTimezone) {
+    await update$dteTimezoneInDB(this._id, this.timezone);
   }
 
-  // Update users settings
+  // TODO: I think the organisationSettings are never updated.
```

Comment:
By this TODO, do you mean that the following updateMany on the user doesn't work?
I think this line can be removed or reverted

Reply:
TODO was removed